### PR TITLE
langchain_google_genai: Update ``_parse_chat_history`` in chat_models.py

### PR DESCRIPTION
**Pull request by andrei-rusu**
- **Description:** Add support for ``ChatMessage`` in ``_parse_chat_history``, making the model compatible with histories that contain this type of message. For example, [streamlit_agent/basic_streaming.py](https://github.com/langchain-ai/streamlit-agent/blob/main/streamlit_agent/basic_streaming.py).
- **Issue:** None existent.
- **Dependencies:** None extra.

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @lkuligin

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 1]
```diff
@@ -314,6 +314,8 @@ def _parse_chat_history(
             role = "model"
         elif isinstance(message, HumanMessage):
             role = "user"
+        elif isinstance(message, ChatMessage):
+            role = message.role
```
Comment (lkuligin) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:318]: what value can `message.role` have? should we maybe map them to "model" / "user" somehow? 
or at least check and throw a meaningful error?

**Comment by lkuligin** [comment]
thanks for adding this!
could you add a unit (and ideally modify an integration) test for that, please?

**Comment by andrei-rusu** [comment]
@lkuligin I am sorry, I've been particularly busy in the past few weeks, so only just coming back to this. To be honest, I am not very familiar with the unit test standards and practices of the repo, as this is my first PR here. Any guidance would be helpful. Thanks!

**Comment by lkuligin** [comment]
thanks for your contribution!

could you send your PR towards https://github.com/langchain-ai/langchain-google, please? these integrations are being moved there. thanks again!

**Comment by baskaryan** [comment]
> thanks for your contribution!
> 
> could you send your PR towards https://github.com/langchain-ai/langchain-google, please? these integrations are being moved there. thanks again!

closing as per @lkuligin's comment

**Pull request closed without merging**